controller/googleOauthHandler.js

```diff
@@ -0,0 +1,102 @@
+require('dotenv').config({
+    path: '../config/variables.env'
+});
+
+const googleOAuth = require('../middleware/googleOAuthClient');
+const User = require('../models/user');
+const tokenGenerator = require('./accessTokenGenerator');
+
+
+
+module.exports.getMessage = async (googleToken) => {
+    try {
+        const response = await googleOAuth.getUserInfo(googleToken);
+        const userInfo = response.data;
+
+        const token = await tokenGenerator.token(userInfo.email);
+        const refresh_token = await tokenGenerator.refresh_token(userInfo.email);
+
+        try {
+            //If the google id already exists in the db, store nothing to the db
+            if (await dbQuery.idExists(userInfo)) {
+                return { 
+                    message: token,
+                    statusCode: 200
+                };
+            } else {
+                //if the google email exists in the db but is not linked, link the accounts
+                if (await dbQuery.emailExists(userInfo)) {
+                    await dbQuery.updateWithId(userInfo);
+                    return {
+                        message: token,
+                        statusCode: 200
+                    };
+                } else {
+                    //otherwise, create a new account
+                    await dbQuery.createAccount(userInfo, refresh_token);
+                    return {
+                        message: token,
+                        statusCode: 201
+                    };
+                }
+            }
+        } catch(error) {
+            console.log(error);
+            return { message:"Error connecting database", statusCode: 500}
+        }
+    } catch(error) {
+        console.log(error);
+        return { message: "BAD_REQUEST", statusCode: 400};
+    }
+};
+
+
+
+const dbQuery = {
+    idExists: async (userInfo) => {
+        try {
+            const googleId = await User.findOne({ google_id: userInfo.id });
+            if (googleId === null) {
```

Comment:
@vipul- directly return ``` return googleId === null ``` its a expression it returns true or false by itself. You don't need to return true or false.